app/helpers/application_helper.rb

```diff
@@ -1,4 +1,16 @@
 module ApplicationHelper
+
+  def liked_by_users(tweet)
+    liked_by = []
+    tweet.likes.each do |user|
+      liked_by << User.find(user.user_id)
+    end
+  end
+
+  def tweet_visitors(tweet)
+    @visitors_are = Visitor.where(tweet_id: tweet.id)
```

Comment:
This should be written in model and not application helper. To know more about helpers and when to use helper-methods, read https://api.rubyonrails.org/classes/ActionController/Helpers.html

Reply:
Okay @swatijadhav thank you